public/app/features/dashboard/dashgrid/PanelChromeAngular.tsx

```diff
@@ -179,9 +183,11 @@ export class PanelChromeAngularUnconnected extends PureComponent<Props, State> {
     const { dashboard, panel, isViewing, isEditing, plugin } = this.props;
     const { errorMessage, data } = this.state;
     const { transparent } = panel;
-
     const alertState = data.alertState?.state;
 
+    const panelChromeProps = getPanelChromeProps(panel, dashboard, data);
+    const panelHeaderTitleItemsProps = panelChromeProps.getPanelHeaderTitleItemsProps();
```

Comment:
👍

Reply:
I'm not sure if I can directly return PanelHeaderTitleItems from the getPanelChromeProps function because it's only a utility JavaScript function and not a React component.

I also considered the alternative of returning the props directly, but it wasn't possible due to some checks involving "showTitleItems" that we need to determine whether to display the title items or not (in this case, panelHeaderTitleItemsProps is null).

```
const getPanelHeaderTitleItemsProps = () => {
    const alertState = data.alertState?.state;
    const showTitleItems =
      (panel.links && panel.links.length > 0 && onShowPanelLinks) ||
      (data.series.length > 0 && data.series.some((v) => (v.meta?.notices?.length ?? 0) > 0)) ||
      (data.request && data.request.timeInfo) ||
      alertState;

    if (!showTitleItems) {
      return null;
    }
 
    return {
      alertState,
      data,
      panelId: panel.id,
      panelLinks: panel.links,
      onShowPanelLinks: onShowPanelLinks,
    };
  };
```

I might be missing a crucial aspect here and potentially complicating the code 😅. If you have any suggestions on how to improve this code, please let me know."

Reply:
@axelavargas I am not sure what the problem is, returning titleItems worked without any issues

https://github.com/grafana/grafana/compare/axelav/migrate-to-angular-panels...migrate-to-angular-panels-title-items?expand=1

Reply:
Ah, I think I see what I was doing wrong. I didn't change the file extension, and my editor was giving errors 😅. I'll apply your suggestions.